flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/linear/HasPredictionDetailCol.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.param.linear;
+
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.StringParam;
+import org.apache.flink.ml.param.WithParams;
+
+/** Interface for the shared prediction detail param. */
+public interface HasPredictionDetailCol<T> extends WithParams<T> {
```

Comment:
Can you explain the use-case for this column? For example, is the value of this column going to be a string? How are the values of this column going to be calculated and interpreted?

Reply:
Logistic regression aims to predict a label for each training data. Note that `predictionCol` is used to report the label for each data point, `PredictionDetailCol` is used to output the probablity of assigning this data point to each class. 

The output should be a double array. For binary classification here, it is a double array with two elements.

Reply:
I see. Since the semantic of this column is the probability of the prediction result, would it be more intuitive to use `HasProbabilityCol` here? The word `detail` is much broader than `probability` and does not provide much information of what is in this column.

Reply:
`HasProbabilityCol ` may not be a very good solution here. I checked for Spark and reused its `HasRawPredictionCol`.

What do you think?

Reply:
Sounds good. Let's use `HasRawPredictionCol`.